pkg/util/file_stats.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package util
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+type FileStats struct {
+	AgentOpenFiles float64 `json:"agent_open_files"`
+	OsFileLimit    float64 `json:"os_file_limit"`
+}
+
+// getFileStats returns the number of files the Agent process has open
+func GetFileStats() *FileStats {
+	stats := FileStats{}
+
+	// Creating a new process.Process type based on Agent PID
+	p, err := process.NewProcess(int32(os.Getpid()))
+	if err != nil {
+		log.Errorf("Failed to retrieve agent process: %s", err)
+		return &stats
+	}
+
+	// Retrieving []OpenFilesStat from Agent process.Process type
+	files, err := p.OpenFiles()
+	if err != nil {
+		log.Errorf("Failed to retrieve agent process' open files slice: %s", err)
+		return &stats
+	}
+
+	// Retrieving number of open files by getting the length of the Agent process.Process type's []OpenFilesStat slice
+	stats.AgentOpenFiles = float64(len(files))
+
+	// Retrieving type []RlimitStat from type process.Process p
+	rs, err := p.Rlimit()
+	if err != nil {
+		log.Errorf("Failed to retrieve type RlimitStat: %s", err)
+		return &stats
+	}
+
+	// Retrieving RLIMIT_NOFILE (index 7) from Agent process' RLimit[]
+	openFilesMaxJSON := []byte(rs[7].String())
```

Comment:
We try to avoid seemingly arbitrary numbers like this in code, the comment above is nice, but in this case we can utilize the `process` packages constants directly.

```suggestion
	openFilesMaxJSON := []byte(rs[process.RLIMIT_NOFILE].String())
```

So that will get rid of the "magic number 7" in here, and also make it resilient to any potential changes in the library.

The next bit of this that is un-necessary is the entire conversion to JSON and back.
The struct returned from `p.Rlimit` has a structure we can use directly, there's no need to convert or parse anything.

```
	softLimit := rs[RLIMIT_NOFILE].Soft
```

That's it! Then, if we switch to use the `RlimitUsage` API, we can also easily get the used amount

```
	used := rs[RLIMIT_NOFILE].Used
```